### PR TITLE
Character moving

### PR DESCRIPTION
**Pull request by agnessaik**
**Sorry a bit bigger pull request than I planned.** 
- Added new class for calculating user visible map data. 
- Added battlefield map some methods. 
- Added eventhandler. 
- and some minor changes.

**Some other issues for the future, ideas are welcome!**
- correct sizing (I didn´t have time to study that. If one of you knows, how to get and calculate window and squares sizes correctly, feel free to correct.)
- optimizing movement handling (I will search that soon, it works but I think it can be way better)
- add methods for sending data regarding to player´s next step to server

### CODE REVIEW

File: client/src/main/java/BattleFieldComponents/BattleFieldMap.java [hunk 1]
```diff
@@ -1,9 +1,35 @@
...
+    }
+
+    public static boolean canGoToSquare(int x, int y) {
+        if (x < 0 || y < 0 || x > battleFieldArray[0].length - 1 || y > battleFieldArray.length - 1){
+            return false;
+        }
+        return SquareTypes.getSquare(battleFieldArray[y][x]).canGoTo();
+    }
+
+    public static int heigth() {
```
Comment (TaliKert) [inline client/src/main/java/BattleFieldComponents/BattleFieldMap.java]: typo > height

File: client/src/main/java/Scenes/LoginAndRegisterScene.java [hunk 2]
```diff
@@ -37,7 +35,6 @@ private GridPane createGridPane() {
         gridPane.setPadding(new Insets(10, 10, 10, 10));
         gridPane.setHgap(10);
         gridPane.setVgap(10);
-
```
Comment (TaliKert) [inline client/src/main/java/Scenes/LoginAndRegisterScene.java]: OCD? :D

File: client/src/main/java/BattleFieldComponents/DisplayLocation.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
+package BattleFieldComponents;
+
+public class DisplayLocation {
+    private static final int SQUARES_IN_ROWS = 11;
+    private static final int SQUARES_IN_COLUMNS = 11;
+
+    private final int userFullMapX;
+    private final int userFullMapY;
```
Comment (TaliKert) [inline client/src/main/java/BattleFieldComponents/DisplayLocation.java]: Namings like this make it really difficult to understand what is meant.

There are a million different map/battlefieldlocation/XYcoordinate centered namings and when they aren't clearly labeled, they can be a headache

Comment (agnessaik) [reply client/src/main/java/BattleFieldComponents/DisplayLocation.java]: I was really struggling to finding good names, to tell if that is location on full map or the little one visible to player, I have worked with these for so long that it all makes sence to me. So if you have any suggestions, please tell.

Comment (TaliKert) [reply client/src/main/java/BattleFieldComponents/DisplayLocation.java]: Maybe I overreacted, sorry :D 
I suggest naming userFullMapX -> playerXCoordinate
same with Y.
DisplayLocation class could be something like RenderedArea, and the instance of it in BattlefieldScene could be renderedArea not userLocation

File: client/src/main/java/BattleFieldComponents/BattleFieldMap.java [hunk 4]
```diff
@@ -1,9 +1,35 @@
 package BattleFieldComponents;
```
[the remark this answers is not in the document]
Comment (KeijoTomiste) [reply client/src/main/java/BattleFieldComponents/BattleFieldMap.java]: package name should match folder name, just saying

File: client/src/main/java/Scenes/LoginAndRegisterScene.java [hunk 5]
```diff
@@ -22,6 +22,11 @@
     private Label infoLabel = new Label();
     private Button loginButton = new Button("Login");
     private Button registerButton = new Button("Register");
+
+    private Disposable isConnectedSubscription;
+
+
+
```
Comment (KeijoTomiste) [inline client/src/main/java/Scenes/LoginAndRegisterScene.java]: 3 line separators is required here, app wont run without

**Pull request merged**